packages/mui-system/src/createStyled/createStyled.js

```diff
@@ -260,6 +244,43 @@ export default function createStyled(input = {}) {
   return styled;
 }
 
+export function applyStyled(props, componentName, overridesResolver) {
+  const styles = [
+    applyThemeOverrides(props, componentName, overridesResolver),
+    // NOTE: Variants only apply for the 'root' slot, not applicable for
+    //       the current use-case.
+    // applyThemeVariants(props, componentName),
+  ]
+  return css(styles);
+}
+
+function applyThemeOverrides(props, componentName, overridesResolver) {
+  const theme = props.theme;
+  const styleOverrides = theme.components?.[componentName]?.styleOverrides;
+  if (!styleOverrides) {
+    return null;
+  }
+
+  const resolvedStyleOverrides = {};
+
+  // TODO: v7 remove iteration and use `resolveStyleArg(styleOverrides[slot])` directly
+  // eslint-disable-next-line guard-for-in
+  for (const slotKey in styleOverrides) {
+    resolvedStyleOverrides[slotKey] = processStyle(props, styleOverrides[slotKey]);
+  }
+
+  return overridesResolver(props, resolvedStyleOverrides);
```

Comment:
@siriwatknp Sidenote but this is part of what can be optimized, right now the whole `styleOverrides` is re-processed each render *for each slot*, even if `overridesResolver` only selects a small slice on the `styleOverrides`. It should be stable as long as `theme` is stable, so we should be able to pre-serialize & memo this. Same for theme variants.